room/compiler-processing/src/test/java/androidx/room/compiler/processing/XTypeElementTest.kt

```diff
@@ -224,8 +237,16 @@ class XTypeElementTest {
                 if (element.isProtected()) result.add("protected")
                 if (element.isPublic()) result.add("public")
                 if (element.isKotlinObject()) result.add("object")
+                if (element.isCompanionObject()) result.add("companion")
+                // Enable once ksp supports fun interfaces - https://github.com/google/ksp/issues/393
+                // if (element.isFunctionalInterface()) result.add("fun")
```

Comment:
what i usually do for these is to still call it (if we don't throw above, we can) then in the assertions, check for `invocation.isKsp` to exclude the check from KSP (so that the javac path is tested)

Reply:
Good call